.github/workflows/fast-dispatch-full-regressions-and-models-impl.yaml

```diff
@@ -155,6 +155,18 @@ jobs:
         if: ${{ matrix.model != 'mistral7b' }}
         run: |
           echo "WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml" >> $GITHUB_ENV
+      - name: Set up llama3.1-8B weight/cache dir (required due to single codebase)
+        if: ${{ matrix.model == 'llama31_8b' }}
```

Comment:
Could we not refactor this into a new function stored somewhere `models/demos/llama3/tests/...` where each individual test for 3.1/3.2-1B/3.2-3B decides what the llama dir is?

You can even use the CI fixture to set it to the appropriate `/mnt/MLPerf`.

Reply:
Not sure if I'm understanding correctly.

Are you suggesting to just have `llama3` on the dispatch .yaml and instead, inside the model folder, parametrize all tests with the `llama_dir`?

or instead having like a `run_ci_dispatch_tests` python function that just calls the other tests?

Reply:
Yes, this is correct.

Right now, I believe you are parameterizing things is by treating the symlink names as the parameter, and passing down arguments from the YAML file. The thing that actually determines which version of llama you're using

Could we instead reorganize things so that this parameterization happens at the pytest level instead? For example, files in `tests/single_card/nightly/llama[version]/` could set this directory env var within pytest code, and call things from `/models/demos/llama3/...`. In any case, we would prefer these folder names embedded into test code, rather than at this level.

Reply:
So the first suggestion

Reply:
what about dealing with this the following way:

Check latest commit diff: https://github.com/tenstorrent/tt-metal/pull/13778/commits/2d8ec834357566d2d49d12730e5ffafc96522d4e

Reply:
GREAT!